block/base/mempool.go

```diff
@@ -155,8 +160,47 @@ func (cm *Mempool[C]) Contains(tx sdk.Tx) bool {
 }
 
 // Compare determines the relative priority of two transactions belonging in the same lane.
-func (cm *Mempool[C]) Compare(ctx sdk.Context, this sdk.Tx, other sdk.Tx) int {
+// There are two cases to consider:
+//  1. The transactions have the same signer. In this case, we compare the sequence numbers.
+//  2. The transactions have different signers. In this case, we compare the priorities of the
+//     transactions.
+//
+// Compare will return -1 if this transaction has a lower priority than the other transaction, 0 if
+// they have the same priority, and 1 if this transaction has a higher priority than the other transaction.
+func (cm *Mempool[C]) Compare(ctx sdk.Context, this sdk.Tx, other sdk.Tx) (int, error) {
+	signers, err := cm.extractor.GetSigners(this)
+	if err != nil {
+		return 0, err
+	}
+	if len(signers) == 0 {
+		return 0, fmt.Errorf("expected one signer for the first transaction")
+	}
+	thisSignerInfo := signers[0]
```

Comment:
By default, the priority nonce mempool always uses the first signer of the tx so this is a safe operation.